src/coreclr/scripts/superpmi.py

```diff
@@ -1824,13 +1830,19 @@ def replay_with_asm_diffs(self):
                             files_with_replay_failures.append(mch_file)
                             result = False
 
+                # Use the MCH file base name as the main part of the directory for generated asm/jitdump/repro artifacts.
+                # Remove the trailing ".mch", and prepend the tag, if any.
+                artifacts_base_name = os.path.basename(mch_file)
+                artifacts_base_name = artifacts_base_name[:-4] if artifacts_base_name.lower().endswith(".mch") else artifacts_base_name
+                artifacts_base_name = "{}.{}".format(self.coreclr_args.tag, artifacts_base_name) if self.coreclr_args.tag is not None else artifacts_base_name
```

Comment:
If we decide to use `-tag` for saving replay failed files, then I would suggest to extract populating the `artifacts_base_name` in a function.